### PR TITLE
Move InvokeGenerator to a centralized function of Meshkit

### LINKED ISSUE #541: Move InvokeGenerator to a centralized function of Meshkit.
**Current Behavior**

When we run `mesheryctl registry generate` we [InvokeGenerator](https://github.com/meshery/meshery/blob/c9728aaaacb4566bf8bc1b2952003083363ae170/mesheryctl/internal/cli/root/registry/generate.go#L144) directly inside the mesheryctl command rather it should be a centralized function in meshkit that can be reused in server as well.

**Desired Behavior**

Update meshkit to contain this functionality.
---
**Resources**

**Alternatives / Additional Context**

### PR DESCRIPTION
**Pull request by dusdjhyeon**
**Description**

This PR fixes #541 

**Notes for Reviewers**
- I moved InvokeGenerator directly into the generators folder, rather than generators/models, due to **circular reference** issues. 
- The functions I moved to utils can now be made to import meshkit from meshery. 
- The terminal output for registry generate in meshery suggests using **io.MultiWriter**. 

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
@Jougan-0 @dusdjhyeon Is there any change in logic?

**Comment by Jougan-0** [comment]
> @Jougan-0 @dusdjhyeon Is there any change in logic?

Nope it was just basic migration.
Next on her list is to update the logs on terminal to show the Models that are being generated in real time and if the program is waiting for an Artifacthub rate limit then show the log of Generation paused for x amount of time for Artifacthub registrant.

**Comment by dusdjhyeon** [comment]
> @dusdjhyeon Did you run this locally? I mean do you have google sheets configured?

I copied meshery Google sheets in my google account and invited my gcp api email and used it.

**Comment by Jougan-0** [comment]
> > @dusdjhyeon Did you run this locally? I mean do you have google sheets configured?
> 
> I copied meshery Google sheets in my google account and invited my gcp api email and used it.

Was everything working fine and models were generated?

**Comment by dusdjhyeon** [comment]
> > > @dusdjhyeon Did you run this locally? I mean do you have google sheets configured?
> > 
> > 
> > I copied meshery Google sheets in my google account and invited my gcp api email and used it.
> 
> Was everything working fine and models were generated?

Yes, I checked. 
image

**Comment by Jougan-0** [comment]
Reviewed on Meshery PR to remove logrus please do so and let's get this merged.

**Comment by leecalcote** [comment]
> Reviewed on Meshery PR to remove logrus please do so and let's get this merged.

👍

File: utils/registry/error.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+func ErrCreateDir(err error, obj string) error {
+	return errors.New(ErrCreateDirCode, errors.Alert, []string{"Error creating directory ", obj}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrGenerateComponent(err error, modelName, compName string) error {
+	return errors.New(ErrGenerateComponentCode, errors.Alert, []string{"error generating comp %s of model %s", compName, modelName}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrFileRead(err error) error {
```
Comment (MUzairS15) [inline utils/registry/error.go:29]: Error already exists, please re-use/enhance.

File: utils/registry/error.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	ErrGenerateModelCode      = "mesheryctl-1055"
+	ErrFileReadCode           = "mesheryctl-1095"
+	ErrMarshalStructToCSVCode = "mesheryctl-1115"
+)
+
+func ErrGenerateModel(err error, modelName string) error {
+	return errors.New(ErrGenerateModelCode, errors.Alert, []string{fmt.Sprintf("error generating model: %s", modelName)}, []string{fmt.Sprintf("Error generating model: %s\n %s", modelName, err.Error())}, []string{"Registrant used for the model is not supported", "Verify the model's source URL.", "Failed to create a local directory in the filesystem for this model."}, []string{"Ensure that each kind of registrant used is a supported kind.", "Ensure correct model source URL is provided and properly formatted.", "Ensure sufficient permissions to allow creation of model directory."})
+}
+
+func ErrCreateDir(err error, obj string) error {
```
Comment (MUzairS15) [inline utils/registry/error.go:21]: Error already exists, please re-use/enhance.

File: utils/registry/error.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package registry
+
+import (
+	"fmt"
+	"github.com/layer5io/meshkit/errors"
+)
+
+var (
+	ErrAppendToSheetCode      = "mesheryctl-1116"
```
Comment (MUzairS15) [inline utils/registry/error.go:9]: Update the error code, please. It's meshkit.

To reduce manual work, replace the code with empty string and run meshkit error utility.

File: utils/registry/error.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+func ErrGenerateModel(err error, modelName string) error {
+	return errors.New(ErrGenerateModelCode, errors.Alert, []string{fmt.Sprintf("error generating model: %s", modelName)}, []string{fmt.Sprintf("Error generating model: %s\n %s", modelName, err.Error())}, []string{"Registrant used for the model is not supported", "Verify the model's source URL.", "Failed to create a local directory in the filesystem for this model."}, []string{"Ensure that each kind of registrant used is a supported kind.", "Ensure correct model source URL is provided and properly formatted.", "Ensure sufficient permissions to allow creation of model directory."})
+}
+
+func ErrCreateDir(err error, obj string) error {
+	return errors.New(ErrCreateDirCode, errors.Alert, []string{"Error creating directory ", obj}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrGenerateComponent(err error, modelName, compName string) error {
```
Comment (MUzairS15) [inline utils/registry/error.go:25]: Check once, this should also exist.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**